conans/test/integration/private_deps_test.py

```diff
@@ -22,8 +22,9 @@ def setUp(self):
         self.client = TestClient(servers=self.servers, users=[("lasote", "mypass")])
 
     def _export_upload(self, name=0, version=None, deps=None, msg=None, static=True):
+        dll_export = self.client.default_compiler_visual_studio and not static
         files = cpp_hello_conan_files(name, version, deps, msg=msg, static=static,
-                                       private_includes=True)
+                                      private_includes=True, dll_export=dll_export)
```

Comment:
The way to handle DLL exports in MSVC is typically to define some macros that define to empty if not MSVC or not dynamic. That way, the code wouldnt need templatizing. But it is ok for now, for testing purposes, not need to modify it, just an observation